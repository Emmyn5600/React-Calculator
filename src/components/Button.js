import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { name, handleClick } = props;

  return (

    <button type="button" className="square" name={name} onClick={handleClick}>

      {name}

    </button>
  );
};

Button.propTypes = {

  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Button;
