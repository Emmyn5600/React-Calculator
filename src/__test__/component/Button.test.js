import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../components/Button';

describe('Button', () => {
  it('should render Buttons', () => {
    const clickHandler = () => '1';
    const tree = renderer.create(<Button name="1" handleClick={clickHandler} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});