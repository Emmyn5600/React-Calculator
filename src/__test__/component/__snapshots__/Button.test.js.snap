// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<button
  name="1"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;

exports[`Button should render Buttons 1`] = `
<button
  name="1"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;

exports[`Button should renders Button 1`] = `
<button
  name="1"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
